fake_switches/juniper/juniper_netconf_datastore.py

```diff
@@ -307,6 +307,19 @@ def parse_vlans(self, conf, etree_conf):
 
         return handled_elements
 
+    def parse_trunk_native_vlan(self, interface_node, port):
+        if len(interface_node.xpath("unit/family/ethernet-switching/native-vlan-id")) == 1 and interface_node.xpath("unit/family/ethernet-switching/native-vlan-id")[0].text is not None:
+            port_attributes = first(interface_node.xpath("unit/family/ethernet-switching"))
+            return resolve_new_value(port_attributes, "native-vlan-id", port.trunk_native_vlan,
+                              transformer=int)
+        return None
+
+    def apply_trunk_native_vlan(self, interface_data, port):
+        if port.vendor_specific["has-ethernet-switching"]:
+            if port.trunk_native_vlan is not None:
+                interface_data[1]['unit']['family']['ethernet-switching']['native-vlan-id'] = str(port.trunk_native_vlan)
```

Comment:
That won't work here's the failing scenario

Try to display an interface with a description and a native vlan and you're screwed

that [1] is assume stuff, what you want it the last interface_data item